benches/indexing_benches.rs

```diff
@@ -0,0 +1,56 @@
+#![feature(test)]
+
+extern crate test;
+
+use cheminee::command_line::indexing::bulk_index::create_tantivy_doc;
+use cheminee::command_line::indexing::split_path;
+use cheminee::indexing::index_manager::IndexManager;
+use cheminee::indexing::KNOWN_DESCRIPTORS;
+use cheminee::schema::LIBRARY;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::ops::Deref;
+use tantivy::schema::Field;
+use test::Bencher;
+
+#[bench]
+fn bench_bulk_index(b: &mut Bencher) {
+    let (storage_dir, index_name) = split_path(&"/tmp/test/benchmark_index".to_string()).unwrap();
+    let index_manager = IndexManager::new(storage_dir.deref(), true).unwrap();
+    let schema = LIBRARY.get("descriptor_v1").unwrap().clone();
+    let index = index_manager
+        .create(index_name.deref(), &schema, true, Some("exactmw"))
+        .unwrap();
+    let mut writer = index.writer(16 * 1024 * 1024).unwrap();
+
+    let smiles_field = schema.get_field("smiles").unwrap();
+    let fingerprint_field = schema.get_field("fingerprint").unwrap();
+    let extra_data_field = schema.get_field("extra_data").unwrap();
+    let descriptor_fields = KNOWN_DESCRIPTORS
+        .iter()
+        .map(|kd| (*kd, schema.get_field(kd).unwrap()))
+        .collect::<HashMap<&str, Field>>();
+
+    let smi =
+        "C[S+](CC[C@@H](C(=O)[O-])[NH3+])C[C@@H]1[C@H]([C@H]([C@@H](O1)N2C=NC3=C(N=CN=C32)N)O)O";
+    let mut map = serde_json::Map::new();
+    map.insert("smiles".to_string(), Value::String(smi.to_string()));
+
+    let record: Value = Value::Object(map);
+
+    b.iter(|| {
+        let doc = create_tantivy_doc(
+            record.clone(),
+            smiles_field,
+            fingerprint_field,
+            &descriptor_fields,
+            extra_data_field,
+        )
+        .unwrap();
+
+        let _ = writer.add_document(doc).unwrap();
+    });
+
+    let _ = writer.commit();
```

Comment:
I think this is a tough thing to benchmark. The I/O really happens in the commit, where any remaining buffered docs are written to disk. I wouldn't try benchmarking I/O, I would stick to, say, searching through docs that are buffered in memory.

Imagine your service going this way:

 1. Cut down size of index based on atom count, etc
 2. Load the top 1000 matching docs in to cheminee's memory
 3. Rank those 1000 matching docs using search routine $XYZ
 4. Return the top N hits

I think you want to create the output vec from step 2 in static memory, somehow, and then you just want to benchmark the serach routine of step 3 and assert the stable sort going in to step 4.

Fake 1/2, benchmark 3, assert 4 is what you expect. If you try to benchmark the actual I/O to get through 1/2 you will find it's highly variable and does not give a reliable picture.